### PR TITLE
Add test for guest setting display name

### PR DESCRIPTION
**Pull request by illicitonion**
Tests for https://github.com/matrix-org/synapse/pull/364

### CODE REVIEW

File: tests/30rooms/13anonymousaccess.pl [hunk 1]
```diff
@@ -333,6 +333,53 @@ sub check_events
...
+               method => "GET",
+               uri    => $displayname_uri,
+            )->then( sub {
+               my ( $body ) = @_;
+               $body->{displayname} eq "creeper" or die "Wrong displayname";
+               Future->done( 1 );
+            }),
+            do_request_json_for( $anonymous_user,
+               method => "GET",
+               uri    => "/api/v1/rooms/$room_id/state/m.room.member/".$anonymous_user->user_id,
```
Comment (NegativeMjark) [inline tests/30rooms/13anonymousaccess.pl]: I think you can use ":user_id" here.

File: tests/30rooms/13anonymousaccess.pl [hunk 2]
```diff
@@ -333,6 +333,53 @@ sub check_events
       });
    };
 
+test "Anonymous user can set display names",
+   requires => [ anonymous_user_fixture(), local_user_and_room_fixtures() ],
+
+   do => sub {
+      my ( $anonymous_user, $user, $room_id ) = @_;
+
+      my $displayname_uri = "/api/v1/profile/".$anonymous_user->user_id."/displayname";
```
Comment (NegativeMjark) [inline tests/30rooms/13anonymousaccess.pl]: You can probably use ``:user_id`` here as well.

File: tests/30rooms/13anonymousaccess.pl (hunk 1 above)
Comment (illicitonion) [inline tests/30rooms/13anonymousaccess.pl]: Done

I do dislike magic, though.

File: tests/30rooms/13anonymousaccess.pl (hunk 2 above)
Comment (illicitonion) [inline tests/30rooms/13anonymousaccess.pl]: Done

**Comment by NegativeMjark** [comment]
LGTM

**Pull request merged**